packages/apex-parser-ast/src/semantics/validation/SObjectRecalculateFormulasValidator.ts

```diff
@@ -0,0 +1,434 @@
+    scope: ValidationScope,
+  ): ValidationResult {
+    const errors: string[] = [];
+    const warnings: string[] = [];
+
+    // Check 1: Error type must be FormulaRecalcFieldError
+    if (!errorInfo.isFormulaRecalcFieldError) {
+      errors.push('method.invalid.recalculate.formulas.wrong.error.type');
+      return {
+        isValid: false,
+        errors,
+        warnings,
+      };
+    }
+
+    // Check 2: Error cannot be null
+    if (errorInfo.isNull) {
+      errors.push('method.invalid.recalculate.formulas.null.error');
+      return {
+        isValid: false,
+        errors,
+        warnings,
+      };
+    }
+
+    return {
+      isValid: errors.length === 0,
+      errors,
+      warnings,
+    };
+  }
+
+  /**
+   * Validates a complete formula recalculation operation
+   *
+   * @param callInfo - Information about the method call
+   * @param resultInfo - Information about the return type
+   * @param scope - Validation scope for error reporting
+   * @returns Validation result indicating if the operation is valid
+   */
+  static validateFormulaRecalculationOperation(
+    callInfo: MethodCallInfo,
+    resultInfo: ReturnTypeInfo,
+    scope: ValidationScope,
+  ): ValidationResult {
+    const errors: string[] = [];
+    const warnings: string[] = [];
+
+    // Validate the method call
+    const callResult = this.validateRecalculateFormulasCall(callInfo, scope);
+    if (!callResult.isValid) {
+      errors.push(...callResult.errors);
+    }
+
+    // Validate the return type
+    const resultValidation = this.validateFormulaRecalcResult(
+      resultInfo,
+      scope,
+    );
+    if (!resultValidation.isValid) {
+      errors.push(...resultValidation.errors);
+    }
+
+    return {
+      isValid: errors.length === 0,
+      errors,
+      warnings,
+    };
+  }
+
+  /**
+   * Check if a parameter is a valid SObject list parameter
+   *
+   * @param parameter - The parameter to check
+   * @returns True if the parameter is a valid SObject list
+   */
+  private static isValidSObjectListParameter(
+    parameter: MethodParameterInfo,
+  ): boolean {
+    // Must be a list
+    if (!parameter.isSObjectList) {
+      return false;
+    }
+
+    // Must be a list type
+    if (!parameter.type.startsWith('List<')) {
+      return false;
+    }
+
+    // Must end with '>'
+    if (!parameter.type.endsWith('>')) {
+      return false;
+    }
+
+    // Extract the inner type
+    const innerType = parameter.type.slice(5, -1); // Remove 'List<' and '>'
+
+    // Must be SObject or a specific SObject type
+    return this.isSObjectType(innerType);
+  }
+
+  /**
+   * Check if a type is an SObject type
+   *
+   * @param type - The type to check
+   * @returns True if the type is an SObject
+   */
+  private static isSObjectType(type: string): boolean {
+    // Direct SObject type
+    if (type === 'SObject') {
+      return true;
+    }
+
+    // Standard SObject types
+    const standardSObjectTypes = [
+      'Account',
```

Comment:
I feel like I've seen this list more than once. Perhaps it can also move into a constant somewhere for reference